### PR TITLE
Add ability to transform phylopics

### PR DESCRIPTION
**Pull request by willgearty**
This adds the ability to rotate and flip phylopics (except for the ability to rotate raster phylopics outside of multiples of 90 degrees, because I have no idea how you would do that). This functionality can be achieved with new exported functions, but has also been incorporated into `add_phylopic()` and `add_phylopic_base()`.

### CODE REVIEW

File: R/phylopic_utils.R [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+#' @family transformations
+#' @export
+flip_phylopic <- function(img, horizontal = TRUE, vertical = FALSE) {
+  if (!is.logical(horizontal)) stop("`horizontal` must be TRUE or FALSE.")
+  if (!is.logical(vertical)) stop("`vertical` must be TRUE or FALSE.")
+  UseMethod("flip_phylopic")
+}
+
+#' @export
+flip_phylopic.Picture <- function(img, horizontal = TRUE, vertical = FALSE) {
```
Comment (LewisAJones) [inline R/phylopic_utils.R:22]: Do we want to include any error handling for `flip_phylopic` and `rotate_phylopic` if they are exported (maybe `recolour_phylopic` too)? I guess testing is already covered by calls in other function tests.

File: R/add_phylopic_base.r [hunk 2]
```diff
@@ -3,28 +3,40 @@
...
+#' @param ysize \code{numeric}. Height of the silhouette. The width is
+#'   determined by the aspect ratio of the original image. Ignored if `x` and
+#'   `y` are not specified.
+#' @param alpha \code{numeric}. A value between 0 and 1, specifying the opacity
+#'   of the silhouette (0 is fully transparent, 1 is fully opaque).
+#' @param color \code{character}. Color to plot the silhouette in.
+#' @param horizontal \code{logical}. Should the silhouette be flipped
+#'   horizontally?
+#' @param vertical \code{logical}. Should the silhouette be flipped vertically?
+#' @param angle \code{numeric}. The number of degrees to rotate the silhouette
```
Comment (LewisAJones) [inline R/add_phylopic_base.r:24]:
```suggestion
#' @param angle \code{numeric}. The number of degrees to rotate the silhouette by
```
Needed?

File: R/add_phylopic.r [hunk 3]
```diff
@@ -5,23 +5,36 @@
...
+#' @param ysize \code{numeric}. Height of the silhouette. The width is
+#'   determined by the aspect ratio of the original image. Ignored if `x` and
+#'   `y` are not specified.
+#' @param alpha \code{numeric}. A value between 0 and 1, specifying the opacity
+#'   of the silhouette (0 is fully transparent, 1 is fully opaque).
+#' @param color \code{character}. Color to plot the silhouette in.
+#' @param horizontal \code{logical}. Should the silhouette be flipped
+#'   horizontally?
+#' @param vertical \code{logical}. Should the silhouette be flipped vertically?
+#' @param angle \code{numeric}. The number of degrees to rotate the silhouette
```
Comment (LewisAJones) [inline R/add_phylopic.r:24]:
```suggestion
#' @param angle \code{numeric}. The number of degrees to rotate the silhouette by
```
Needed?

**Review by LewisAJones — APPROVED** [review]
Hi @willgearty, 

This all looks pretty great to me. The additional functionality behaves exactly as expected and I think the transformations are easy for the user to follow. I have a couple of small comments but other than that, things look great! Thanks for getting that together.

File: R/add_phylopic.r (hunk 3 above)
Comment (willgearty) [reply R/add_phylopic.r:24]: If you want to be grammatically correct with "by", you'd need to say "The number of degrees by which to rotate the silhouette", and that just sounds like too much to me.

**Pull request merged**